src/agent/no-full-response.ts

```diff
@@ -0,0 +1,75 @@
+// agent/no-full-response.ts
+
+/*
+ * Copyright (c) 2021-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import { ESLintUtils, TSESTree } from '@typescript-eslint/utils';
+import { strict as assert } from 'node:assert';
+import getDocumentationUrl from '../get-documentation-url';
+import { getTypeParentNode } from '../library/ts-tree';
+
+export const ruleId = 'no-full-response';
+
+const createRule = ESLintUtils.RuleCreator((name) => getDocumentationUrl(name));
+
+const rule = createRule({
+  name: ruleId,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Remove the usage of FullResponse type.',
+    },
+    messages: {
+      removeFullResponse: 'Removing the usage of FullResponse type.',
+      unknownError: 'Unknown error occurred in file "{{fileName}}": {{ error }}.',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const sourceCode = context.sourceCode;
+
+    return {
+      'TSTypeReference[typeName.name="FullResponse"]': (typeReference: TSESTree.TSTypeReference) => {
+        try {
+          const typeParentNode = getTypeParentNode(typeReference);
+          assert.ok(typeParentNode);
+          if (typeParentNode.type === TSESTree.AST_NODE_TYPES.TSAsExpression) {
+            context.report({
+              messageId: 'removeFullResponse',
+              node: typeReference,
+              fix(fixer) {
+                return fixer.replaceText(typeParentNode, sourceCode.getText(typeParentNode.expression));
+              },
+            });
+          } else {
+            context.report({
+              messageId: 'removeFullResponse',
+              node: typeReference,
+              fix(fixer) {
+                return fixer.remove(typeParentNode);
+              },
+            });
+          }
+        } catch (error) {
+          // eslint-disable-next-line no-console
```

Comment:
Not a big deal, just wondering if we need this console.error?  I suppose I don't really understand what circumstances could lead to this error.